controllers/v1/observationsController.js

```diff
@@ -1079,13 +1082,49 @@ module.exports = class Observations extends Abstract {
                 }
                 let observationDocument = await observationsHelper.observationDocuments( queryObject )
                 observationDocument = observationDocument[0]
+                
                 if (!observationDocument) {
                     return resolve({ 
                         status: httpStatusCode.bad_request.status, 
                         message: messageConstants.apiResponses.OBSERVATION_NOT_FOUND
                     });
                 }
-
+                // join observation's program. PII data consent is given via this api call.
+                // check if already joined the program
+                //Check data present in programUsers collection.
+                const programUsers = await programUsersHelper.programUsersDocuments(
+                    {
+                        userId : req.userDetails.userId,
+                        programId : observationDocument.programId
+                    },
+                    ["_id"]
+                );
+                
+                if ( !programUsers.length > 0 && ( appVersion === "" || appVersion < 5.2 ) ) {
```

Comment:
@aks30 can we have a common variable PIIFeatureReleaseVersion and use it.

Reply:
No version-specific coding, is there no other way? Please plan a discussion if we have to

Reply:
@VISHNUDAS-tunerlabs  we can add validation if requestForPIIConsent is true and user has not joined program then will call program join api.

Reply:
@ankitpws join api call anyway we have to do to increment noOfResourcesStarted. because as @aks30 suggested below comment create/update have to happen via core service. same think we have to change for projects

Reply:
@VISHNUDAS-tunerlabs  we can have an increment process via Kafka or we can directly update in DB (right now we are using an update method). we can remove this checking app version will check only for if user has joined program then will do nothing and if he is not joined then will call program Join API in core services.